best_practices/controllers.rst

```diff
@@ -164,13 +177,14 @@ manually. In our application, we have this situation in ``CommentController``:
 
 .. code-block:: php
 
+    use Doctrine\ORM\EntityManagerInterface;
+
     /**
      * @Route("/comment/{postSlug}/new", name = "comment_new")
      */
-    public function newAction(Request $request, $postSlug)
+    public function newAction(Request $request, $postSlug, EntityManagerInterface $em)
```

Comment:
> EntityManagerInterface looks and feels terrible

We may ultimately decide in the future to allow non-interface autowiring where it makes sense - e.g. simply `EntityManager $em`.

> $this->getDoctrine() is not especially discouraged whereas $this->get() is

We can definitely change whatever the recommendation is in the future. But, as @GuilhemN said, there's really nothing wrong with `$this->getDoctrine()` or the other shortcut methods. The advantage to `getDoctrine()` is discoverability (I get auto-complete with `$this->getD`). The disadvantage is that it's not transferrable to a service, which is something we're trying to reduce (i.e. make controllers look more like services... while balancing DX). 

I'm quite mixed on which to use...